update-db.js

```diff
@@ -117,66 +100,100 @@ function diffBrowsersLists (old, current) {
   if (node.dependencies) {
-    delete node.dependencies['caniuse-lite']
+    if (node.dependencies['caniuse-lite']) {
+      var version = node.dependencies['caniuse-lite'].version
+      metadata.versions[version] = true
+      delete node.dependencies['caniuse-lite']
+    }
     for (var i in node.dependencies) {
-      node.dependencies[i] = deletePackage(node.dependencies[i])
+      node.dependencies[i] = deletePackage(node.dependencies[i], metadata)
     }
   }
   return node
 }
 
-module.exports = function updateDB (print) {
-  var lock = detectLockfile()
+var yarnVersionRe = new RegExp('version "(.*?)"')
+
+function updateYarnLockfile (lock, latest) {
+  var blocks = lock.content.split(/(\n{2,})/).map(function (block) {
+    return block.split('\n')
+  })
+  var versions = {}
+  blocks.forEach(function (lines) {
+    if (lines[0].indexOf('caniuse-lite@') !== -1) {
+      var match = yarnVersionRe.exec(lines[1])
+      versions[match[1]] = true
+      if (match[1] !== latest.version) {
+        lines[1] = lines[1].replace(
+          /version "[^"]+"/, 'version "' + latest.version + '"'
+        )
+        lines[2] = lines[2].replace(
+          /resolved "[^"]+"/, 'resolved "' + latest.dist.tarball + '"'
+        )
+        lines[3] = lines[3].replace(
+          /integrity .+/, 'integrity ' + latest.dist.integrity
+        )
+      }
+    }
+  })
+  var content = blocks.map(function (lines) {
+    return lines.join('\n')
+  }).join('')
+  return { content: content, versions: versions }
+}
+
+function updatePnpmLockfile (lock, latest) {
+  var versions = {}
+  var caniusePackage
+  var content = yaml.safeLoad(lock.content)
+
+  for (var pkg in content.packages) {
+    if (pkg.indexOf('/caniuse-lite') >= 0) {
+      // delete the node and replace it with the new data
+      caniusePackage = content.packages[pkg]
+      var version = pkg.replace('/caniuse-lite/', '')
+      versions[version] = true
+      delete content.packages[pkg]
+    }
+    if (content.packages[pkg] &&
+      content.packages[pkg].dependencies &&
+      content.packages[pkg].dependencies['caniuse-lite']) {
+      versions[content.packages[pkg].dependencies['caniuse-lite']] = true
+      content.packages[pkg].dependencies['caniuse-lite'] = latest.version
+    }
+  }
+
+  if (caniusePackage) {
+    caniusePackage.resolution.integrity = latest.dist.integrity
+    content.packages['/caniuse-lite/' + latest.version] = caniusePackage
+  }
+  return { content: yaml.safeDump(content), versions: versions }
+}
+
+function updateLockfile (lock, latest) {
+  // read the lock file
   lock.content = fs.readFileSync(lock.file).toString()
 
-  var current = getCurrentVersion(lock)
+  if (lock.mode === 'npm') {
+    return updateNpmLockfile(lock, latest)
+  } else if (lock.mode === 'yarn') {
+    return updateYarnLockfile(lock, latest)
+  }
+  return updatePnpmLockfile(lock, latest)
+}
+
+module.exports = function updateDB (print) {
+  print('running updateDb!\n')
```

Comment:
Is the next command slow? If so, it will be better to explain what happening like `Reading lock file…`

Reply:
Removed.